feat(bkp) : Add backup service to do the bkp (INTLY-2482) (#125)
aerogear/mobile-security-service-operator: README.adoc

NOTE: The command `make monitoring/uninstall` will uninstall the Monitor Service.

=== Configuring the Backup Service

==== Backup

===== Install
The backup service is implemented by using  https://github.com/integr8ly/backup-container-image[integr8ly/backup-container-image]. It will do the backup of the database and the APP CRs which are data required to be restore in the case of failures. Following the steps to enable it.

> . Setup the AWS in order to store the backup outside of the cluster. You need to add your AWS details to link:./mobile-security-service-operator/deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservicebackup_cr.yaml[MobileSecurityServiceBackup] as follows or add the name of the secret which has already this data in the cluster.

Reviewer: This doesn't read right

Author: I did not understood your questions.

Author: Sorry,  I did not get your question. 
The AWS details are required to create the Secret to send the file AWS S3. However, it will be not used by the RHMI. Just for the community devs and/or other product which could decide impl an integration with.

NOTE: The command `make monitoring/uninstall` will uninstall the Monitor Service.

=== Storing Backup in AWS

> The credentials details to store the backup in your AWS are setup in the link:./mobile-security-service-operator/deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservicebackup_cr.yaml[MobileSecurityServiceBackup] as follows.

Reviewer: ```suggestion
To enable the backup storage on AWS you need to add your AWS credentials to  link:./mobile-security-service-operator/deploy/crds/mobile-security-service_v1alpha1_mobilesecurityservicebackup_cr.yaml[MobileSecurityServiceBackup] as follows.
```

Author: 👍

NOTE: The command `make monitoring/uninstall` will uninstall the Monitor Service.

> === Configuring the Backup Service

Author: @laurafitzgerald now you have the required info to start your PR.

Author: closing it since shows sorted.

+
[source,yaml]
----
  # ---------------------------------
  # Stored Host - AWS
  # ----------------------------

  awsS3BucketName: "example-awsS3BucketName"
  awsAccessKeyId: "example-awsAccessKeyId"
  awsSecretAccessKey: "example-awsSecretAccessKey"
----
+
. Run the command `make backup/install` in the same namespace where the MobileSecurityService Database is installed in order to apply the CronJob which will do this process.
+
IMPORTANT: The namespaces are setup manually in the role link:./deploy/backup/cluster_role_binding[cluster_role_binding]. You should replace them if the operator is not installed in the default namespace.
+
. Add the label `integreatly-middleware-service=true` in all namespaces defined in the ENV VAR APP_NAMESPACES in the link:./deploy/operator.yaml[operator.yaml]. (E.g  oc label ns/mobile-security-service-apps integreatly-middleware-service=true`). It will backup all link:./deploy/crds/examples/mobile-security-service_v1alpha1_mobilesecurityserviceapp_cr.yaml[MobileSecurityServiceApp CR] applied on them.

> ==== Restore

Author: @laurafitzgerald @matskiv here has the steps. We can address the SOP to here.

Reviewer 2: 👍 thanks @camilamacedo86 I've been addressing those also in [this pr](https://github.com/fheng/integreatly-help/pull/173) but I think that it's good to have it here in this README in the case of a non intly install.

Author: Closing it since shows sorted.

| `SDKConfigMapName` | Name of the config map object created for the app with its SDK Config.
|===

* link:./pkg/apis/mobilesecurityservice/v1alpha1/mobilesecurityservicebackup_types.go[MobileSecurityServiceBackup]
+
|===
| *Status*    | *Description*

> | `backupStatus` | Should shows `OK` when all was created with success.

Reviewer: ```suggestion
| `backupStatus` | Should show `OK` when everything is created successfully.
```

Author: 👍

| `SDKConfigMapName` | Name of the config map object created for the app with its SDK Config.
|===

> * link:./pkg/apis/mobilesecurityservice/v1alpha1/mobilesecurityservicebackup_types.go[MobileSecurityServiceBackup]

Author: @laurafitzgerald it is the info that should be checked by the admin (SOP)

Author: Closing it since shows sorted.
